frontend/packages/knative-plugin/src/topology/components/groups/KnativeServiceGroup.tsx

```diff
@@ -110,6 +108,17 @@ const KnativeServiceGroup: React.FC<KnativeServiceGroupProps> = ({
     }
   }, [editAccess, innerHover, onShowCreateConnector, onHideCreateConnector, allowEdgeCreation]);
 
+  const decorators = getNodeDecorators(
+    element,
+    element.getGraph().getData().decorators,
+    x + width / 2,
+    y + height / 2,
+    -1,
+    DECORATOR_RADIUS,
+    width,
+    height,
+  );
+
```

Comment:
Convert to a component `<Decorators ...`

In there you can use `useResolvedExtensions` to fetch the decorators instead of setting them in the graph data.

Reply:
I believe it would be too expensive to resolve extensions (and sort, etc) for every node instance.

Reply:
After reviewing `useResolvedExtensions`, there is no cache yet. Brought this up with Vojtech to improve.

Still value in the Decorators component but we can tackle it later.